scripts/versioning/check.ts

```diff
@@ -0,0 +1,38 @@
+// this script will be triggered from workflow actions when a new tag is push
+import {Version} from '../../typechain'
+import isSemver from 'is-semver'
+
+import {log} from '../../config/logging'
+
+/*
+ * takes a contract that inherits Version and returns true if the contract's version is equal to the release tag
+ * throws otherwise
+ *
+ * @param A Contract object that inherits Version
+ *
+ */
+export async function checkContractVersionAgainstReleaseTag(
+    contract: Version
+): Promise<boolean> {
+    const gitSourceTag = process.env.SOURCE_TAG
+
+    if (gitSourceTag) {
+        log.info(`git tag: ${gitSourceTag}`)
+    } else {
+        log.info(`git tag not defined!`)
+        process.exit(1)
+    }
+    // validate syntax of the git tag - could we do this instead in husky? before commit reaches remote/origin?
```

Comment:
@CjHare

Reply:
No. Husky cannot be relied upon. Husky can fail to install in certain setups and is not used in the CI/CD.

In this PR you've added https://www.npmjs.com/package/is-semver as a dependency, can't you use that to validate whether the input is semver?